### ISSUE TITLE
Question regarding rbind2 when combining flowsets with different column order

### ISSUE THREAD

**Issue by gcohenJH** [RGLab/flowCore#219]
Hi, I want to combine two flowsets (or cytosets) into one, and it seems like `rbind2` is the only way to do so. The columns in my flowsets are in different orders though, which results in an error. If it helps, I imported the data from a flowjo file using CytoML.
```
>colnames(fs1)
 [1] "Time"                   "SSC-H"                  "SSC-A"                  "FSC-H"                 
 [5] "FSC-A"                  "SSC-B-H"                "SSC-B-A"                "Comp-BV421-A"          
 [9] "Comp-BV480-A"           "Comp-BV510-A"           "Comp-BV570-A"           "Comp-BV650-A"          
[13] "Comp-BV711-A"           "Comp-BV785-A"           "Comp-Alexa Fluor 488-A" "Comp-Spark Blue 550-A" 
[17] "Comp-PE-Dazzle594-A"    "Comp-PE-Cy5-A"          "Comp-PE-Cy7-A"          "Comp-APC-A"            
[21] "Comp-Alexa Fluor 647-A" "Comp-Zombie NIR-A"      "Comp-APC-Fire 750-A"    "Comp-APC-Fire 810-A"   
>colnames(fs2)
 [1] "FSC-A"                  "FSC-H"                  "SSC-A"                  "SSC-B-A"               
 [5] "SSC-B-H"                "SSC-H"                  "Comp-APC-A"             "Comp-APC-Fire 750-A"   
 [9] "Comp-APC-Fire 810-A"    "Comp-Alexa Fluor 488-A" "Comp-Alexa Fluor 647-A" "Comp-BV421-A"          
[13] "Comp-BV480-A"           "Comp-BV510-A"           "Comp-BV570-A"           "Comp-BV650-A"          
[17] "Comp-BV711-A"           "Comp-BV785-A"           "Comp-PE-Cy5-A"          "Comp-PE-Cy7-A"         
[21] "Comp-PE-Dazzle594-A"    "Comp-Spark Blue 550-A"  "Comp-Zombie NIR-A"      "Time"  
> all(colnames(fs1) %in% colnames(fs2))
[1] TRUE
> all(colnames(fs2) %in% colnames(fs1))
[1] TRUE
>rbind2(fs1, fs2)
<sample name> doesn't have the identical colnames as the other samples!
....
```
`rbind2` works if I sort the column names or order them to be identical, but as you might expect, it results in data and descriptions being mismatched.
A simple recreation using the GvHD data.
```
library(flowCore)
data(GvHD)
fs1 <- GvHD[1:4]
fs2 <- GvHD[5:8]

#Works correctly with matching column order
rbind2(fs1, fs2)
#> A flowSet with 8 experiments.
#> 
#> An object of class 'AnnotatedDataFrame'
#>   rowNames: s5a01 s5a02 ... s6a01 (8 total)
#>   varLabels: Patient Visit ... name (5 total)
#>   varMetadata: labelDescription
#> 
#> column names(8): FSC-H SSC-H ... FL4-H Time

#Though this would lead to mismatched data after binding it should still work
colnames(fs2)[c(1, 8)] <- colnames(fs2)[c(8, 1)]
rbind2(fs1, fs2)
#> s5a05 doesn't have the identical colnames as the other samples!
#> s5a06 doesn't have the identical colnames as the other samples!
#> s5a07 doesn't have the identical colnames as the other samples!
#> s6a01 doesn't have the identical colnames as the other samples!
#> Error in validObject(.Object): invalid class "flowSet" object: Some items identified in the data environment either have the wrong dimension or type.
```
I cannot find a function to reorder the columns in a flowset, flowframe, cytoset, or cytoframe so they match and work with `rbind2`. Is there a way to do so, or is there another method to combine flowsets that does not have this error?

**Comment by SamGG** [comment]
Hi.
Interesting question. A flowSet allows one to select columns as for a classical matrix. Let's imagine you realign fs2 to fs1. So the code below should work, or inspire you. Let us know if it did.
Best
```
new_order = match(colnames(fs2), colnames(fs1))
fs2_reordered = fs2[,new_order]
fs_all = rbind2(fs1, fs2_reordered)
```

**Comment by gcohenJH** [comment]
Thanks Sam; that worked! (Though switch fs1 and fs2 in the first line)
I thought I had tried something similar, but it had resulted in the column names getting mismatched from their data (using `fs#[, order(colnames(fs#)]`). I just tried that now and it also worked. I think I had tried renaming the columns using `colnames() <-` in an earlier attempt and forgot to revert it (which resulted in anything I did being mismatched).
